src/main/java/net/snowflake/ingest/streaming/internal/RowBufferStats.java

```diff
@@ -131,6 +161,9 @@ void addStrValue(String value) {
 
   void addBinaryValue(byte[] valueBytes) {
     this.setCurrentMaxLength(valueBytes.length);
+    if (enableDistinctValue) {
+      distinctValues.add(new String(Base64.getEncoder().encode(valueBytes)));
```

Comment:
ouch this will kill performance - both cpu and memory. If I'm not wrong we are already doing heavy validation of input strings, I'm hoping we can reuse that single pass on the whole string.

Another idea is to keep a `Map<String, Set<Int>>` here for string columns, the map key can be a CRC32 hash and the map value will be a list of row ids that contain a value with that hash. Only in the case where the Set has size > 1 do we look at the actual strings, before serialization. This is non-trivial amount of changes all over though.

Reply:
Discussed offline. Lets keep NDV disabled by default. When it's enabled, skip the string ndv computation. Added a [JIRA](https://snowflakecomputing.atlassian.net/browse/SNOW-1721193) for this.